src/geoserver/role_service.py

```diff
@@ -0,0 +1,97 @@
+from distutils.dir_util import copy_tree
+import logging
+import os
+import shutil
+import sys
+import time
+from urllib.parse import urlparse
+from lxml import etree as ET
+
+from db import util as db_util
+from requests_util import url_util
+from . import authn
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+
+
+ROLE_SERVICE_PATH = 'security/role/'
+DIRECTORY = os.path.dirname(os.path.abspath(__file__))
+
+
+def wait_for_db(conn_dict):
+    max_attempts = 10
+    attempt = 0
+
+    while True:
+        import psycopg2
+        try:
+            with psycopg2.connect(**conn_dict):
+                pass
+            logger.info(f"  Attempt {attempt}/{max_attempts} successful.")
+            break
+        except psycopg2.OperationalError:
+            if attempt >= max_attempts:
+                logger.info(f"  Reaching max attempts when waiting for DB")
+                sys.exit(1)
+            time.sleep(2)
+            attempt += 1
+
+
+def setup_role_service(data_dir, db_conn, uri_str, internal_service_schema, layman_pg_user, layman_gs_user, layman_gs_role, service_url, role_service_name):
+    logger.info(f"Ensuring GeoServer DB role service '{role_service_name}' "
+                f"for URL: {service_url}.")
+
+    logger.info(f"  Waiting for DB")
+    wait_for_db(db_conn)
+
+    logger.info(f"  Checking internal role service DB schema")
+    schema_query = f'''SELECT COUNT(*) FROM information_schema.schemata WHERE schema_name = '{internal_service_schema}';'''
+    schema_exists = db_util.run_query(schema_query, uri_str=uri_str)[0][0]
+    if schema_exists == 0:
+        logger.info(f"    Setting up internal role service DB schema")
+        statement = f"""
+    CREATE SCHEMA "{internal_service_schema}" AUTHORIZATION {layman_pg_user};
+    create view {internal_service_schema}.roles as select 'ADMIN' as name, null as parent
+    union all select 'GROUP_ADMIN', null
+    union all select 'LAYMAN_ROLE', null
+    ;
+    create view {internal_service_schema}.role_props as select null::varchar as rolename, null::varchar as propname, null::varchar as propvalue;
+    create view {internal_service_schema}.user_roles as select 'layman' as username, 'ADMIN' as rolename
+    union all select 'layman', 'LAYMAN_ROLE'
+    union all select 'admin', 'ADMIN'
```

Comment:
Within ensure_admin_roles in admin_user_roles view add also GEOSERVER_ADMIN_USER, 'ADMIN' relation